load-tests/src/load-test.ts

```diff
@@ -0,0 +1,284 @@
+    check(data, {
+      "model id is valid": (data) => Number.isInteger(data.insert_mission_model_one.id)
+    });
+
+    sleep(5);
+  });
+
+}
+
+export function query_missions() {
+  const query = `#graphql
+    query {
+      mission_model {
+        id
+      }
+    }
+  `;
+
+  const data = req.hasura(query, {}, token, username);
+  check(data, {
+    "data is nonnull": (data) => data !== null,
+    "id is valid": (data) => Number.isInteger(data.mission_model[0].id)
+  });
+
+  sleep(Math.random() * 5);
+}
+
+export function insert_plans() {
+  const query = `#graphql
+    query {
+      mission_model {
+        id
+      }
+    }
+  `;
+
+  const rd = Math.random() * 10000;
+
+  const data = req.hasura(query, {}, token, username);
+  const { mission_model } = data;
+  const ids: number[] = mission_model;
+  const id = ids[Math.floor(Math.random() * ids.length)].id;
+
+  const input: CreatePlanInput = {
+    model_id: id,
+    name: 'test_plan' + rd + id,
+    start_time: plan_start_timestamp,
+    duration: "1d"
+  };
+
+  const resp = req.hasura(gql.CREATE_PLAN, {plan: input}, token, username);
+
+  sleep(Math.random() * 5);
+}
+
+export function insert_random_activities() {
+  const query = `#graphql
+    query {
+      plan {
+        id
+      }
+    }
+  `;
+
+  const rd = Math.random() * 10000;
+
+  const data = req.hasura(query, {}, token, username);
+  const { plan } = data;
+  const ids: number[] = plan;
+  const random_id = ids[Math.floor(Math.random() * ids.length)];
+
+  if (!random_id) {
+    sleep(5);
+    return;
+  }
+
+  const random_hour_offset = Math.floor(Math.random() * 24);
+
+  const input: ActivityInsertInput = {
+      arguments: {
+        biteSize: 1,
+      },
+      plan_id: id,
+      type: 'BiteBanana',
+      start_offset: random_hour_offset + 'h',
+  };
+
+  const resp = req.hasura(gql.CREATE_ACTIVITY_DIRECTIVE, {activityDirectiveInsertInput: input}, token, username);
+
+  sleep(Math.random() * 5);
+}
+
+export function insert_targeted_activities() {
+
+  const random_seconds_offset = Math.floor(Math.random() * 24 * 60 * 60);
+
+  const input: ActivityInsertInput = {
+      arguments: {
+        biteSize: 1,
+      },
+      plan_id: 1,
+      type: 'BiteBanana',
+      start_offset: random_seconds_offset + 's',
+  };
+
+  const resp = req.hasura(gql.CREATE_ACTIVITY_DIRECTIVE, {activityDirectiveInsertInput: input}, token, username);
+
+  sleep(Math.random() * 5);
+}
+
+export function run_targeted_simulations() {
+  const resp = req.hasura(gql.SIMULATE, {plan_id: 1}, token, username);
+  sleep(Math.random() * 30);
+}
+
+export function run_random_simulations() {
+  const query = `#graphql
+    query {
+      plan {
+        id
+      }
+    }
+  `;
+
+  const data = req.hasura(query, {}, token, username);
+  const { plan } = data;
+  const ids: number[] = plan;
+  const random_id = ids[Math.floor(Math.random() * ids.length)];
```

Comment:
In that case, `random_id` will be `undefined` which we check for in the next line.

Reply:
Can you alter the if statement so that if `ids` is empty, the method returns prior to setting `random_id`? That is actually what's being checked with `if(random_id)`

Reply:
I've refactored this logic a bit, and I think I address your issue now.

Reply:
I've added checks to catch several different errors now, which will show up in our statistics, rather than placing several `if === undefined` in the test functions themselves. I.e. we _want_ to fail loud in these scenarios, not gracefully handle errors.

How is the `number[]` type cast an error?

Reply:
> I've added checks to catch several different errors now, which will show up in our statistics, rather than placing several if === undefined in the test functions themselves

I believe when I wrote this comment there were `!== undefined` checks in the test functions. The current error catching is good 👍 

> How is the `number[]` type cast an error?

This bit:

```ts
const random_plan = plan_ids[...];
return random_plan.id;
```

`plan_ids` is defined as `number[]` (which isn't the type of `data.plan`), which means that `random_plan` is of type `number`. The type `number` has no properties. Therefore, the fact that the code accesses the `.id` property of`random_plan` means that it isn't actually a `number`.